src/urllib3/exceptions.py

```diff
@@ -14,6 +14,11 @@ class HTTPWarning(Warning):
     pass
 
 
+class DisconnectError(Exception):
```

Comment:
I understand this isn't strictly an HTTP Error but I think we have everything sub-class that so that users can catch any urllib3 exception handily and this breaks that assumption severely.